server/athenian/api/__init__.py

```diff
@@ -385,5 +391,16 @@ def main() -> Optional[AthenianApp]:
         ui=args.ui, auth0_cls=auth0_cls, kms_cls=kms_cls, cache=cache, slack=slack,
         client_max_size=int(os.getenv("ATHENIAN_MAX_CLIENT_SIZE", 256 * 1024)),
         max_load=float(os.getenv("ATHENIAN_MAX_LOAD", 12)))
+    if args.preload_dataframes:
+        app.on_dbs_conected(partial(_preload_dataframes, log))
     app.run(host=args.host, port=args.port, print=lambda s: log.info("\n" + s))
     return app
+
+
+def _preload_dataframes(log, dbs, fut):
+    log.info("Preparing MemoryCache")
+    mdb, pdb = dbs["mdb"], dbs["pdb"]
+    options = build_memory_cache_options(mdb, pdb)
+    mc = MemoryCache.get_instance(mdb, pdb, options)
+    task = asyncio.create_task(mc.load())
+    task.add_done_callback(lambda *_: log.info("MemoryCache READY"))
```

Comment:
We should think about graceful destruction here: if the app crashes before the task finishes, we'll end with "task pending but destroyed"-like errors. The easiest seems to be registering a callback on application shutdown: https://docs.aiohttp.org/en/stable/web_advanced.html#graceful-shutdown

How about making a `DataFramePreloader` class and putting code as his methods? I feel that the code will nicer.

Reply:
Agreed, I'll add a `DataFramePreloader` 👍